File: backend/app/models/player.py
from pydantic import BaseModel
from typing import List, Optional


class PlayerStats(BaseModel):
    player: str
    season: str
    position: str
    age: int
    team: str
    games_played: int
    games_started: int
    minutes_per_game: float
    points_per_game: float
    field_goals_per_game: float
    field_goal_attempts_per_game: float
    field_goal_percentage: float
    three_pointers_per_game: float
    three_point_attempts_per_game: float
    three_point_percentage: float
    two_pointers_per_game: float
    two_point_attempts_per_game: float
    two_point_percentage: float
    effective_field_goal_percentage: float
    free_throws_per_game: float
    free_throw_attempts_per_game: float
    free_throw_percentage: float
    offensive_rebounds_per_game: float
    defensive_rebounds_per_game: float
    total_rebounds_per_game: float
    assists_per_game: float
    steals_per_game: float
    blocks_per_game: float
    turnovers_per_game: float
    personal_fouls_per_game: float
    player_id: str


class SimilarPlayer(BaseModel):
    player: str
    season: str
    position: str
    age: int
    similarity_score: float
    stats: PlayerStats


class SimilarPlayersResponse(BaseModel):
    query_player: PlayerStats
    similar_players: List[SimilarPlayer]


class PlayerQuery(BaseModel):
    player_name: str
    season: str = "2023_24"
    num_similar: int = 5

File: frontend/types/player.ts
export interface PlayerStats {
  player: string;
  season: string;
  position: string;
  age: number;
  team: string;
  games_played: number;
  games_started: number;
  minutes_per_game: number;
  points_per_game: number;
  field_goals_per_game: number;
  field_goal_attempts_per_game: number;
  field_goal_percentage: number;
  three_pointers_per_game: number;
  three_point_attempts_per_game: number;
  three_point_percentage: number;
  two_pointers_per_game: number;
  two_point_attempts_per_game: number;
  two_point_percentage: number;
  effective_field_goal_percentage: number;
  free_throws_per_game: number;
  free_throw_attempts_per_game: number;
  free_throw_percentage: number;
  offensive_rebounds_per_game: number;
  defensive_rebounds_per_game: number;
  total_rebounds_per_game: number;
  assists_per_game: number;
  steals_per_game: number;
  blocks_per_game: number;
  turnovers_per_game: number;
  personal_fouls_per_game: number;
  player_id: string;
}

export interface SimilarPlayer {
  player: string;
  season: string;
  position: string;
  age: number;
  similarity_score: number;
  stats: PlayerStats;
}

export interface SimilarPlayersResponse {
  query_player: PlayerStats;
  similar_players: SimilarPlayer[];
}

export interface PlayerSkillRatings {
  scoring: number;
  playmaking: number;
  defense: number;
  athleticism: number;
  basketball_iq: number;
}

export interface PlayerWithSkillRatings extends PlayerStats {
  skillRatings: PlayerSkillRatings;
}

export interface SimilarPlayerWithSkillRatings extends SimilarPlayer {
  skillRatings: PlayerSkillRatings;
}
